test/units/module_utils/facts/network/test_generic_bsd.py

```diff
@@ -64,152 +48,173 @@ def get_bin_path(command, warning=None):
-    if command == 'fake/route':
-        return 0, 'Foo', ''
-    if command == ['fake/ifconfig', '-a']:
-        return 0, netbsd_ifconfig_a_out_7_1, ''
-    return 1, '', ''
-
-
-def run_command_post_7_1_ifconfig(command):
-    if command == 'fake/route':
-        return 0, 'Foo', ''
-    if command == ['fake/ifconfig', '-a']:
-        return 0, netbsd_ifconfig_a_out_post_7_1, ''
-    return 1, '', ''
-
-
-class TestGenericBsdNetworkNetBSD(unittest.TestCase):
-    gather_subset = ['all']
-
-    def setUp(self):
-        self.maxDiff = None
-        self.longMessage = True
-
-    # TODO: extract module run_command/get_bin_path usage to methods I can mock without mocking all of run_command
-    def test(self):
-        module = self._mock_module()
-        module.get_bin_path.side_effect = get_bin_path
-        module.run_command.side_effect = run_command_old_ifconfig
-
-        bsd_net = generic_bsd.GenericBsdIfconfigNetwork(module)
-
-        res = bsd_net.populate()
-        self.assertDictEqual(res, NETBSD_EXPECTED)
-
-    def test_ifconfig_post_7_1(self):
-        module = self._mock_module()
-        module.get_bin_path.side_effect = get_bin_path
-        module.run_command.side_effect = run_command_post_7_1_ifconfig
-
-        bsd_net = generic_bsd.GenericBsdIfconfigNetwork(module)
-
-        res = bsd_net.populate()
-        self.assertDictEqual(res, NETBSD_EXPECTED)
-
-    def test_netbsd_ifconfig_old_and_new(self):
-        module_new = self._mock_module()
-        module_new.get_bin_path.side_effect = get_bin_path
-        module_new.run_command.side_effect = run_command_post_7_1_ifconfig
-
-        bsd_net_new = generic_bsd.GenericBsdIfconfigNetwork(module_new)
-        res_new = bsd_net_new.populate()
-
-        module_old = self._mock_module()
-        module_old.get_bin_path.side_effect = get_bin_path
-        module_old.run_command.side_effect = run_command_old_ifconfig
-
-        bsd_net_old = generic_bsd.GenericBsdIfconfigNetwork(module_old)
-        res_old = bsd_net_old.populate()
-
-        self.assertDictEqual(res_old, res_new)
-        self.assertDictEqual(res_old, NETBSD_EXPECTED)
-        self.assertDictEqual(res_new, NETBSD_EXPECTED)
-
-    def _mock_module(self):
-        mock_module = Mock()
-        mock_module.params = {'gather_subset': self.gather_subset,
-                              'gather_timeout': 5,
-                              'filter': '*'}
-        mock_module.get_bin_path = Mock(return_value=None)
-        return mock_module
-
-    def test_ensure_correct_netmask_parsing(self):
-        n = generic_bsd.GenericBsdIfconfigNetwork(None)
-        lines = [
-            'inet 192.168.7.113 netmask 0xffffff00 broadcast 192.168.7.255',
-            'inet 10.109.188.206 --> 10.109.188.206 netmask 0xffffe000',
-        ]
-        expected = [
+"""
+
+NETBSD_EXPECTED = {
+    "all_ipv4_addresses": ["192.168.122.205"],
+    "all_ipv6_addresses": ["fe80::5054:ff:fe63:55af%re0"],
```

Comment:
I like the pytest updates. I personally would prefer to not change all the ' to " - unless it's enforced since it seems like it could just be undone similarly, and refactoring adds unnecessary noise to git blame.

Reply:
ok

Reply:
you missed this one still